photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/LinearRegressionOptimizationProblemTest.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.optimization
+
+import com.linkedin.photon.ml.data.{LabeledPoint, SimpleObjectProvider}
+import com.linkedin.photon.ml.model.Coefficients
+import com.linkedin.photon.ml.normalization.NormalizationContext
+import com.linkedin.photon.ml.optimization.game.GLMOptimizationConfiguration
+import com.linkedin.photon.ml.test.CommonTestUtils
+
+import org.apache.spark.rdd.RDD
+import org.mockito.Mockito._
+import org.testng.Assert._
+import org.testng.annotations.Test
+
+class LinearRegressionOptimizationProblemTest {
+  import LinearRegressionOptimizationProblemTest._
+  import CommonTestUtils._
+
+  @Test
+  def testUpdateObjective(): Unit = {
+    val problem = createProblem
+    val normalizationContext = new SimpleObjectProvider(mock(classOf[NormalizationContext]))
+    val regularizationWeight = 1D
+
+    assertNotEquals(problem.regularizationWeight, regularizationWeight)
+
+    val updatedProblem = problem.updateObjective(normalizationContext, regularizationWeight)
+
+    assertEquals(updatedProblem.regularizationWeight, regularizationWeight)
+  }
+
+  @Test
+  def testInitializeZeroModel(): Unit = {
+    val problem = createProblem
+
+    val dim = 10
+    val zeroModel = problem.initializeZeroModel(dim)
+
+    assertEquals(zeroModel.coefficients, Coefficients.initializeZeroCoefficients(dim))
+  }
+
+  @Test
+  def testCreateModel(): Unit = {
+    val problem = createProblem
+
+    val dim = 10
+    val coefficients = generateDenseVector(dim)
+    val model = problem.createModel(coefficients, None)
+
+    assertEquals(model.coefficients.means, coefficients)
+  }
+
+  @Test
+  def testComputeVariances(): Unit = {
+    val problem = createProblem
+
+    val input = mock(classOf[RDD[LabeledPoint]])
+    val dim = 10
+    val coefficients = generateDenseVector(dim)
+
+    // TODO: computeVarainces is currently disabled. This test will need to be updated when the default changes
+    assertEquals(problem.computeVariances(input, coefficients), None)
```

Comment:
We should have 2 tests, one for the None case and one that verifies correct variance computation.

This comment and the one above it apply to all 4 of the GLM OP descendant tests.